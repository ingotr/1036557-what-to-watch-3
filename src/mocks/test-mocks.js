const FirstMovie = {
  TITLE: `Mock-test-The Grand Budapest Hotel`,
  GENRE: `Mock-test-Drama`,
  RELEASE_DATE: 2020,
};

const movies = [
  {
    id: `mock-test-011`,
    title: `Fantastic Beasts: The Crimes of Grindelwald`,
    imgSrc: `img/dardjeeling-limited.jpg`,
    previewSrc: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
  },
  {
    id: `mock-test-012`,
    title: `Bohemian Rhapsody`,
    imgSrc: `img/dardjeeling-limited.jpg`,
    previewSrc: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
  },
  {
    id: `mock-test-013`,
    title: `Macbeth`,
    imgSrc: `img/dardjeeling-limited.jpg`,
    previewSrc: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
  },
  {
    id: `mock-test-014`,
    title: `Aviator`,
    imgSrc: `img/dardjeeling-limited.jpg`,
    previewSrc: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
  },
  {
    id: `mock-test-015`,
    title: `We need to talk about Kevin`,
    imgSrc: `img/dardjeeling-limited.jpg`,
    previewSrc: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
  },
  {
    id: `mock-test-016`,
    title: `What We Do in the Shadows`,
    imgSrc: `img/dardjeeling-limited.jpg`,
    previewSrc: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
  },
  {
    id: `mock-test-017`,
    title: `Revenant`,
    imgSrc: `img/dardjeeling-limited.jpg`,
    previewSrc: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
  },
  {
    id: `mock-test-018`,
    title: `Johnny English`,
    imgSrc: `img/dardjeeling-limited.jpg`,
    previewSrc: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
  },
];

const movieInfo = {
  title: `Mock-test-The Grand Budapest Hotel`,
  genre: `Mock-test-Drama`,
  year: 2017,
  poster: {
    big: `img/the-grand-budapest-hotel-poster.jpg`,
    bigAlt: `Mock-test-The Grand Budapest Hotel poster`,
  },
  rating: {
    score: 8.9,
    level: `Mocks-Very good`,
    count: 240,
  },
  director: `Mock-Wes Andreson`,
  description: [`Mock-test-In the 1930s, the Grand Budapest Hotel is a popular European ski resort, presided over by concierge Gustave H. (Ralph Fiennes). Zero, a junior lobby boy, becomes Gustave &#96s friend and protege.`,
    `Gustave prides himself on providing first-className service to the hotel &#96s guests, including satisfying the sexual needs of the many elderly women who stay there. When one of Gustave &#96s lovers dies mysteriously, Gustave finds himself the recipient of a priceless painting and the chief suspect in her murder.`],
  starring: [`Mock-test-Bill Murray`, `Edward Norton`, `Jude Law`, `Willem Dafoe`],
};

export {FirstMovie, movies, movieInfo};
