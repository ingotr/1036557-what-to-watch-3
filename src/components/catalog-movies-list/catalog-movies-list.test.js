import React from 'react';
import renderer from 'react-test-renderer';
import CatalogMoviesList from './catalog-movies-list.jsx';

import {movies} from '../../mocks/test-mocks.js';

it(`Render App`, () => {
  const tree = renderer
   .create(<CatalogMoviesList
     movies={movies}
     onMovieHover={() => {}}
   />)
   .toJSON();

  expect(tree).toMatchSnapshot();
});
