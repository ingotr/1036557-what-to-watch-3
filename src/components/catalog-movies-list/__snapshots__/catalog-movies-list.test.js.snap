// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render App 1`] = `
<div
  className="catalog__movies-list"
>
  <article
    className="small-movie-card catalog__movies-card"
  >
    <video
      className="player__video--play"
      disabled={true}
      muted="muted"
      onMouseOut={[Function]}
      onMouseOver={[Function]}
      poster="img/dardjeeling-limited.jpg"
      src="https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4"
      width="250"
    />
    <h3
      className="small-movie-card__title"
    >
      <a
        className="small-movie-card__link"
        href="movie-page.html"
      >
        Fantastic Beasts: The Crimes of Grindelwald
      </a>
    </h3>
  </article>
  <article
    className="small-movie-card catalog__movies-card"
  >
    <video
      className="player__video--play"
      disabled={true}
      muted="muted"
      onMouseOut={[Function]}
      onMouseOver={[Function]}
      poster="img/dardjeeling-limited.jpg"
      src="https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4"
      width="250"
    />
    <h3
      className="small-movie-card__title"
    >
      <a
        className="small-movie-card__link"
        href="movie-page.html"
      >
        Bohemian Rhapsody
      </a>
    </h3>
  </article>
  <article
    className="small-movie-card catalog__movies-card"
  >
    <video
      className="player__video--play"
      disabled={true}
      muted="muted"
      onMouseOut={[Function]}
      onMouseOver={[Function]}
      poster="img/dardjeeling-limited.jpg"
      src="https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4"
      width="250"
    />
    <h3
      className="small-movie-card__title"
    >
      <a
        className="small-movie-card__link"
        href="movie-page.html"
      >
        Macbeth
      </a>
    </h3>
  </article>
  <article
    className="small-movie-card catalog__movies-card"
  >
    <video
      className="player__video--play"
      disabled={true}
      muted="muted"
      onMouseOut={[Function]}
      onMouseOver={[Function]}
      poster="img/dardjeeling-limited.jpg"
      src="https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4"
      width="250"
    />
    <h3
      className="small-movie-card__title"
    >
      <a
        className="small-movie-card__link"
        href="movie-page.html"
      >
        Aviator
      </a>
    </h3>
  </article>
  <article
    className="small-movie-card catalog__movies-card"
  >
    <video
      className="player__video--play"
      disabled={true}
      muted="muted"
      onMouseOut={[Function]}
      onMouseOver={[Function]}
      poster="img/dardjeeling-limited.jpg"
      src="https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4"
      width="250"
    />
    <h3
      className="small-movie-card__title"
    >
      <a
        className="small-movie-card__link"
        href="movie-page.html"
      >
        We need to talk about Kevin
      </a>
    </h3>
  </article>
  <article
    className="small-movie-card catalog__movies-card"
  >
    <video
      className="player__video--play"
      disabled={true}
      muted="muted"
      onMouseOut={[Function]}
      onMouseOver={[Function]}
      poster="img/dardjeeling-limited.jpg"
      src="https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4"
      width="250"
    />
    <h3
      className="small-movie-card__title"
    >
      <a
        className="small-movie-card__link"
        href="movie-page.html"
      >
        What We Do in the Shadows
      </a>
    </h3>
  </article>
  <article
    className="small-movie-card catalog__movies-card"
  >
    <video
      className="player__video--play"
      disabled={true}
      muted="muted"
      onMouseOut={[Function]}
      onMouseOver={[Function]}
      poster="img/dardjeeling-limited.jpg"
      src="https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4"
      width="250"
    />
    <h3
      className="small-movie-card__title"
    >
      <a
        className="small-movie-card__link"
        href="movie-page.html"
      >
        Revenant
      </a>
    </h3>
  </article>
  <article
    className="small-movie-card catalog__movies-card"
  >
    <video
      className="player__video--play"
      disabled={true}
      muted="muted"
      onMouseOut={[Function]}
      onMouseOver={[Function]}
      poster="img/dardjeeling-limited.jpg"
      src="https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4"
      width="250"
    />
    <h3
      className="small-movie-card__title"
    >
      <a
        className="small-movie-card__link"
        href="movie-page.html"
      >
        Johnny English
      </a>
    </h3>
  </article>
</div>
`;
